modules/encrypt/classes/Kohana/Encrypt/Engine/Openssl.php

```diff
@@ -10,83 +9,50 @@
  *
  * The Cipher
  * :  A [cipher](http://php.net/manual/en/openssl.ciphers.php) determines how the encryption
- *    is mathematically calculated. By default, the "AES-256-CBC" cipher
- *    is used.
+ *    is mathematically calculated. By default, the "AES-256-CBC" cipher is used.
+ *
  * @package    Kohana/Encrypt
  * @category   Security
- * @author     Kohana Team
- * @copyright  (c) Kohana Team
+ * @author     Koseven Team
+ * @copyright  (c) 2007-2012 Kohana Team
+ * @copyright  (c) 2016-2018 Koseven Team
  * @license    https://koseven.ga/LICENSE.md
  */
 class Kohana_Encrypt_Engine_Openssl extends Kohana_Encrypt_Engine
 {
-    use Traits_Encrypt_Iv;
-
-    /** @var string AES_128_CBC code */
+	/**
+	 * AES_128_CBC
+	 * @var	string
+	 */
     const AES_128_CBC = 'AES-128-CBC';
-    /** @var string AES_256_CBC code */
-    const AES_256_CBC = 'AES-256-CBC';
-
-    /**
-     * @var string Engine type
-     */
-    const TYPE = 'Openssl';
 
-    /**
-     * @var int the size of the Initialization Vector (IV) in bytes
-     */
-    protected $_iv_size;
+	/**
+	 * AES_256_CBC
+	 * @var	string
+	 */
+    const AES_256_CBC = 'AES-256-CBC';
 
     /**
      * Creates a new openssl wrapper.
-     *
-     * @param array $config Array with configuration
+     * @param  array $config Configuration
      * @throws Kohana_Exception
      */
     public function __construct(array $config)
     {
-        if (!extension_loaded('openssl'))
+        if ( ! extension_loaded('openssl'))
         {
             throw new Kohana_Exception('OpenSSL extension is not installed.');
         }
 
         parent::__construct($config);
 
-        if (empty($config[self::CONFIG_CIPHER]))
-        {
-            // Add the default cipher
-            $this->_cipher = static::AES_256_CBC;
-        }
-        else
-        {
-            $this->_cipher = $config[self::CONFIG_CIPHER];
-        }
+		$this->_cipher = $config['cipher'] ?? self::AES_256_CBC;
```

Comment:
Hmm actually wondering if this is the same thing as $config['mode'] in mcrypt?

Reply:
No. according to documentation and code here is the difference:
'cipher' determines how the encryption is mathematically calculated e.g: AES-128
'mode' etermines how the encrypted data is written in binary form e.g: nofb